### ISSUE TITLE
Fix all test case

### ISSUE THREAD

**Issue by KitFung** [jhelovuo/RustDDS#111]
Since I may want  to use this in my project, I am reviewing the codes and trying to make it more stable.
I found that the code base in master branch cannot pass the testcase.

Moreover, it may be great for adding some guideline about how to contribute (which toolchain version? the roadmap? etc)

```
failures:

---- dds::reader::tests::rtpsreader_handle_gap stdout ----
thread 'dds::reader::tests::rtpsreader_handle_gap' panicked at 'called `Option::unwrap()` on a `None` value', src/dds/reader.rs:303:53

---- dds::reader::tests::rtpsreader_handle_data stdout ----
thread 'dds::reader::tests::rtpsreader_handle_data' panicked at 'assertion failed: rec.try_recv().is_ok()', src/dds/reader.rs:1083:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- dds::reader::tests::rtpsreader_notification stdout ----
thread 'dds::reader::tests::rtpsreader_notification' panicked at 'assertion failed: rec.try_recv().is_ok()', src/dds/reader.rs:1024:5

---- structure::dds_cache::tests::create_dds_cache stdout ----
thread 'structure::dds_cache::tests::create_dds_cache' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `3`', src/structure/dds_cache.rs:355:5

---- structure::inline_qos::tests::inline_qos_key_hash stdout ----
thread 'structure::inline_qos::tests::inline_qos_key_hash' panicked at 'called `Result::unwrap()` on an `Err` value: Eof', src/structure/inline_qos.rs:143:51


failures:
    dds::reader::tests::rtpsreader_handle_data
    dds::reader::tests::rtpsreader_handle_gap
    dds::reader::tests::rtpsreader_notification
    structure::dds_cache::tests::create_dds_cache
    structure::inline_qos::tests::inline_qos_key_hash

```

**Comment by jhelovuo** [comment]
We use latest stable toolchain.

There is no definite roadmap, but the current production branch is 0.4.x.

The next major upgrade is 0.5, which will 
* rename some functions and otherwise improve good Rust style
* add derive macro to detect key length, which should improve dispose interoperability

**Comment by jhelovuo** [comment]
Tests are "fixed" in 0.5.1 by disabling broken tests for now.
